### PR TITLE
Work on BLL and API has been started

### PR DESCRIPTION
**Pull request by RainbowInHell**
*The first DTO entity has been added;
*An appropriate mapper has been created;
*Created service for the DTO entity;

*Added controller and Startup class.

### CODE REVIEW

**Review by KatsiarynaAnashkevich — APPROVED** [review]

File: CarRental/Extensions/RegistrarExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿using CarRental.Registrars;
+using CarRental.Registrars.Contracts;
+
+namespace CarRental.Extensions
+{
+    public static class RegistrarExtensions
```
Comment (MaksimSilkou) [inline CarRental/Extensions/RegistrarExtensions.cs:6]: i think, you have hire typo

File: CarRental/Extensions/RegistrarExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            foreach (var registrar in registrars)
+            {
+                registrar.RegisterServices(builder);
+            }
+        }
+
+        public static void RegisterPipelineComponents(this WebApplication app, Type scanningType)
+        {
+            var registrars = GetRegistrars<IWebApplicationRegistrar>(scanningType);
+            foreach (var registrar in registrars)
```
Comment (MaksimSilkou) [inline CarRental/Extensions/RegistrarExtensions.cs:21]: enter before return

File: CarRental/Registrars/Contracts/IRegistrar.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿namespace CarRental.Registrars.Contracts
+{
+    public interface IRegistrar
+    {
```
Comment (MaksimSilkou) [inline CarRental/Registrars/Contracts/IRegistrar.cs:4]: Why an empty interface?

**Review by kirillturok — APPROVED** [review]

File: CarRental/Registrars/Contracts/IRegistrar.cs (hunk 3 above)
Comment (RainbowInHell) [reply CarRental/Registrars/Contracts/IRegistrar.cs:4]: It's a marker interface, to indicate that a class implements a certain behavior or functionality. There is no need to add any actual functionality to the interface itself. For certain functionality i have IWebApplicationBuilderRegistrar and IWebApplicationRegistrar interfaces.

File: CarRental/Extensions/RegistrarExtensions.cs (hunk 2 above)
Comment (RainbowInHell) [reply CarRental/Extensions/RegistrarExtensions.cs:21]: *

File: CarRental/Extensions/RegistrarExtensions.cs (hunk 1 above)
Comment (RainbowInHell) [reply CarRental/Extensions/RegistrarExtensions.cs:6]: Where exactly? I don't get it.

**Pull request merged**